Sources/BuildSystemIntegration/DetermineBuildSystem.swift

```diff
@@ -49,24 +51,26 @@ package func determineBuildSystem(
     return nil
   }
   for buildSystemType in buildSystemPreference {
+    var spec: BuildSystemSpec? = nil
+
     switch buildSystemType {
     case .buildServer:
-      if let projectRoot = ExternalBuildSystemAdapter.projectRoot(for: workspaceFolderUrl, options: options) {
-        return BuildSystemSpec(kind: .buildServer, projectRoot: projectRoot)
-      }
+      spec = ExternalBuildSystemAdapter.searchForConfig(
+        in: workspaceFolderUrl,
+        onlyConsiderRoot: onlyConsiderRoot,
+        options: options
+      )
     case .compilationDatabase:
-      if let projectRoot = CompilationDatabaseBuildSystem.projectRoot(for: workspaceFolderUrl, options: options) {
-        return BuildSystemSpec(kind: .compilationDatabase, projectRoot: projectRoot)
-      }
+      spec = CompilationDatabaseBuildSystem.searchForConfig(in: workspaceFolderUrl, options: options)
     case .swiftPM:
       #if canImport(PackageModel)
-      if let projectRoot = SwiftPMBuildSystem.projectRoot(for: workspaceFolderUrl, options: options) {
-        return BuildSystemSpec(kind: .swiftPM, projectRoot: projectRoot)
-      }
-      #else
-      return nil
+      spec = SwiftPMBuildSystem.searchForConfig(in: workspaceFolderUrl, options: options)
       #endif
     }
+
+    if spec != nil {
+      return spec
+    }
```

Comment:
```suggestion
    if let spec {
      return spec
    }
```
 
Just looks a little more swifty.